backend/app/ai/utils/ai_search.py

```
@@ -31,54 +31,93 @@
         json_text = re.search(r"{.*}", text, re.DOTALL)
         if json_text:
             return json.loads(json_text.group())
         else:
+            return {"keywords": [], "keywords_en": [], "synonyms": {}}
+    except Exception as e:
+        return {"keywords": [], "keywords_en": [], "synonyms": {}}
 
-            return {"keywords": [], "keywords_en": []}
-    except Exception:
-        return {"keywords": [], "keywords_en": []}
+def extract_price_range_from_text(text: str) -> Union[List[float], None]:
```

Ruff (0.8.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)